### DISCUSSION TITLE
Why deleting model event not fired when use whereIn with pivot query?

### DISCUSSION THREAD

**Discussion by MykolaKovrizhnykh** [laravel/framework#33911] [General] [answered]
![image](https://user-images.githubusercontent.com/66183919/90407899-b9d71280-e0af-11ea-843c-2499d388e324.png)

**Comment by GrahamCampbell** [comment, marked as answer]
That is because a delete query is not the same as a model delete. It bypasses the model.

**Reply by MykolaKovrizhnykh** [reply]
@GrahamCampbell 

But I use pivot model like this

![image](https://user-images.githubusercontent.com/66183919/90426822-56a6a980-e0ca-11ea-93f1-e7dcee75fccd.png)

And tried to detach with where clause (is is customized sync function)

![image](https://user-images.githubusercontent.com/66183919/90426910-7a69ef80-e0ca-11ea-95e3-8323dd85660e.png)


But because I use where clause deleting event is not firing. Maybe needs to add $this->fireModelEvent when using pivot model?
